### ISSUE TITLE
ng serve not working.

### ISSUE THREAD

**Issue by esafwan** [angular/angular-cli#2844]
> Please provide us with the following information:
> ---------------------------------------------------------------

### OS?
>macOs Sierra

### Versions.
>Can't  run `ng --version`. It's returning error. 

### Repro steps.

First ng new command was not working. The below error was there.
> Cannot find module 'reflect-metadata'.
I then uninstalled angular-cli, Installed 'reflect-metadata' module & then had to install 'port-finder' module too, both globally. After this I cleared npm cache and re-installed 'angular-cli'. Now, 'ng new' command works. Bug ng serve doesn't work. It's again saying 'reflect-metadata' module is missing. 

### The log given by the failure.

> Error when ng serve is used

`MacBook-Pro$ ng serve
Cannot find module 'reflect-metadata'
Error: Cannot find module 'reflect-metadata'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/@ngtools/webpack/src/index.js:5:1)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/models/webpack-build-typescript.js:5:17)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/models/webpack-config.js:2:34)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/tasks/build-webpack.js:6:24)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/commands/build.js:3:23)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Class.module.exports.includedCommands (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/addon/index.js:21:16)
    at /Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/lib/models/project.js:392:61
    at Array.forEach (native)
    at Project.addonCommands (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/lib/models/project.js:391:15)
    at Project.eachAddonCommand (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/lib/models/project.js:426:30)
    at module.exports (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/lib/cli/lookup-command.js:33:13)
    at CLI.<anonymous> (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/lib/cli/cli.js:34:26)
    at tryCatch (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:538:12)
    at invokeCallback (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:553:13)
    at publish (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:521:7)
    at flush (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:2373:5)
    at nextTickCallbackWith0Args (node.js:420:9)
    at process._tickCallback (node.js:349:13)`

> Error when ng --version is used. I installed angular-cli today. Not able to get the version no. as command is not working. 

`MacBook-Pro:$ ng --version
Cannot find module 'reflect-metadata'
Error: Cannot find module 'reflect-metadata'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/@ngtools/webpack/src/index.js:5:1)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/models/webpack-build-typescript.js:5:17)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/models/webpack-config.js:2:34)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/tasks/build-webpack.js:6:24)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/commands/build.js:3:23)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Class.module.exports.includedCommands (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/addon/index.js:21:16)
    at /Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/lib/models/project.js:392:61
    at Array.forEach (native)
    at Project.addonCommands (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/lib/models/project.js:391:15)
    at Project.eachAddonCommand (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/lib/models/project.js:426:30)
    at module.exports (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/lib/cli/lookup-command.js:33:13)
    at CLI.<anonymous> (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/lib/cli/cli.js:34:26)
    at tryCatch (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:538:12)
    at invokeCallback (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:553:13)
    at publish (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:521:7)
    at flush (/Users/safwan/Code/www/angular/head/node_modules/angular-cli/node_modules/angular-cli/node_modules/rsvp/dist/rsvp.js:2373:5)
    at nextTickCallbackWith0Args (node.js:420:9)
    at process._tickCallback (node.js:349:13)`

**Comment by nadickens** [comment]
I had similar errors but I have just got it working.

I was on a low version of node and npm, so installed the latest version of node (6.9.1) and did the following:
npm i -g reflect-metadata
npm i -g portfinder

npm uninstall -g angular-cli
npm cache clean 
npm install -g angular-cli

ng new <project>
cd <project>
ng server

And it worked.

**Comment by esafwan** [comment]
- I had installed node from https://nodejs.org/en/, directly downloading the binary. 
- In the site it might be a but confusing as it shows recommended for most users against the version 4.6.1 LTS binary. 
- I manually downloaded the one that says v6.9.1 and installed it in my mac without uninstalling the old one. 
- Then I followed the steps mentioned above by @nadickens 
- The issue still exist on the old project I created. But its no more there in any new projects i create with `ng new`.

Screenshot from nodejs website given below:
<img width="651" alt="screen shot 2016-10-25 at 1 44 55 am" src="https://cloud.githubusercontent.com/assets/246454/19662343/af84c2d2-9a54-11e6-86f3-378570c1f263.png">
